Remotion/ObjectBinding/Web/UI/Controls/Validation/IValidationFailureMatcher.cs

```diff
@@ -0,0 +1,30 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using System.Collections.Generic;
+using Remotion.ObjectBinding.Validation;
+
+namespace Remotion.ObjectBinding.Web.UI.Controls.Validation
+{
+  public interface IValidationFailureMatcher
+  {
+    /// <summary>
+    /// Defines the required API to determine validation errors relevant to a <see cref="BocList"/> row.
+    /// </summary>
+    IReadOnlyCollection<BusinessObjectValidationFailure> GetMatchingValidationFailures (IBusinessObject rowObject, IBusinessObjectValidationResult validationResult);
```

Comment:
Naming: since the IBocSortableColumnDef uses "Create' not "Get" , we should do that here, too.

Reply:
There is nothing to create here as this only collects the failures already created elsewhere. 

Perhaps you were referring to `IBocColumnWithValidationSupport`?

Otherwise I do not see the added benefit of having this copy the text of the SortableColumnDef as it just doesn't fit in this context.

Reply:
Gah, I licked the wrong line. Yes, I meant IBocColumnWithValidationSupport